vcd/resource_vcd_external_network_v2.go

```diff
@@ -0,0 +1,465 @@
+package vcd
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+var networkV2IpScope = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"gateway": &schema.Schema{
+			Type:         schema.TypeString,
+			Required:     true,
+			Description:  "Gateway of the network",
+			ValidateFunc: validation.IsIPAddress,
+		},
+		"prefix_length": &schema.Schema{
+			Type:         schema.TypeInt,
+			Required:     true,
+			Description:  "Network mask",
+			ValidateFunc: validation.IntAtLeast(1),
+		},
+		"enabled": &schema.Schema{
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+			Description: "If subnet is enabled",
+		},
+		"static_ip_pool": &schema.Schema{
```

Comment:
Quick question. I tried to change `end_address` from `14.14.14.25` to `14.14.14.26` of a _single_ **IP pool**, but _all_ IP **scopes** showed an update. Is this expected?
```
# vcd_external_network_v2.ext-net-nsxt will be updated in-place
  ~ resource "vcd_external_network_v2" "ext-net-nsxt" {
        description = "First NSX-T backed network"
        id          = "urn:vcloud:network:5ea10a81-41bf-4ff2-a8b3-e56d843d4b38"
        name        = "nsxt-network"

      - ip_scope {
          - enabled       = false -> null
          - gateway       = "88.88.88.1" -> null
          - prefix_length = 24 -> null

          - static_ip_pool {
              - end_address   = "88.88.88.100" -> null
              - start_address = "88.88.88.88" -> null
            }
        }
      - ip_scope {
          - enabled       = true -> null
          - gateway       = "14.14.14.1" -> null
          - prefix_length = 24 -> null

          - static_ip_pool {
              - end_address   = "14.14.14.15" -> null
              - start_address = "14.14.14.10" -> null
            }
          - static_ip_pool {
              - end_address   = "14.14.14.25" -> null
              - start_address = "14.14.14.20" -> null
            }
        }
      + ip_scope {
          + enabled       = true
          + gateway       = "14.14.14.1"
          + prefix_length = 24

          + static_ip_pool {
              + end_address   = "14.14.14.15"
              + start_address = "14.14.14.10"
            }
          + static_ip_pool {
              + end_address   = "14.14.14.26"
              + start_address = "14.14.14.20"
            }
        }
      + ip_scope {
          + enabled       = false
          + gateway       = "88.88.88.1"
          + prefix_length = 24

          + static_ip_pool {
              + end_address   = "88.88.88.100"
              + start_address = "88.88.88.88"
            }
        }

        nsxt_network {
            nsxt_manager_id      = "urn:vcloud:nsxtmanager:ebfcf8f8-b4a0-4389-ae53-2ce4900ec699"
            nsxt_tier0_router_id = "router-tier-0"
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

Reply:
Yeah, can't do much here. It is because these are nested `TypeSet`s and children changing TypeSet hash results in parents chancing hashes. Diff is shown like that then.